File: Dersler/Programalama Dillerinin Prensipleri/Ödevler/Diğer Ödevler/Programlama Dillerinin Prensipleri Yaptığım 3.Ödev/KoloniSavasi/src/AUretim.java
/**
 * @author Muhammet Kemal Güvenç --- kemal.guvenc@ogr.sakarya.edu.tr
 * @since 22.05.2023
 * <p>
 * AUretim Sınıfı: Kolonilerin kullandığı A Üretim tekniğini tanımlar.
 * </p>
 */

public class AUretim implements Uretim {
    /**
     * Bu üretim taktiğinde koloninin popülasyon ve yemek değerleri birbirinden çıkarılır.
     * Bu değerin 10'u geçmemesi için 11'le modu alınır.
     *
     * @param populasyon Koloninin popülasyonu
     * @param yemek      Koloninin yemek miktarı
     * @return Koloninin ürettiği yemek miktarı
     */
    @Override
    public int Uret(int populasyon, int yemek) {
        int deger = (populasyon - yemek) % 11;

        return deger;
    }
}

File: Dersler/Programlama Dillerinin Prensipleri/Ödevler/Diğer Ödevler/Programlama Dillerinin Prensipleri Yaptığım 2.Ödev/src/AUretim.c
/*
 * @author Muhammet Kemal Güvenç --- kemal.guvenc@ogr.sakarya.edu.tr
 * @since 21.05.2023
 * AUretim yapısının kaynak kodları
 */

#include "../include/AUretim.h"

AUretim AUretimOlustur() {
    AUretim this;
    this = malloc(sizeof(struct AURETIM));
    this -> super = UretimOlustur();
    this -> super -> Uret = &AUret;
    this -> AUretimYokEt = &AUretimYokEt;

    return this;
}

void AUretimYokEt(AUretim const this) {
    if (this == NULL)
        return;

    this -> super -> UretimYokEt(this -> super);
    free(this);
}

// Bu üretim taktiğinde koloninin popülasyon ve yemek değerleri birbirinden çıkarılır.
// Bu değerin 10'u geçmemesi için 10'la modu alınır.
int AUret(AUretim const this, int populasyon, int yemek) {
    int deger = (populasyon - yemek) % 10;

    return deger;
}
